scripts/normalize-by-median.py

```diff
@@ -344,6 +344,11 @@ def main():  # pylint: disable=too-many-branches,too-many-statements
         else:
             output_name = args.single_output_file.name
         outfp = args.single_output_file
```

Comment:
str(fileobj) doesn't actually work in a useful way.

```
In [1]: f = open('file', 'w')

In [2]: str(f)
Out[2]: "<open file 'file', mode 'w' at 0x7f703005a4b0>"
```

On Fri, Jul 24, 2015 at 12:24 PM C. Titus Brown <notifications@github.com>
wrote:

> In scripts/normalize-by-median.py
> <https://github.com/dib-lab/khmer/pull/1186#discussion_r35438920>:
>
> > @@ -344,6 +344,11 @@ def main():  # pylint: disable=too-many-branches,too-many-statements
> >          else:
> >              output_name = args.single_output_file.name
> >          outfp = args.single_output_file
>
> The logic seems reasonably straightforward to me and I don't see how to
> simplify it easily. I can't tell if I'm misreading it or if you're
> misreading it :). Help?
>
> Also note that .name doesn't exist in certain cases that are hit by the
> tests. @bocajnotnef <https://github.com/bocajnotnef> has been dealing
> with this by just applying str(args.single_output_file), which I guess
> works.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dib-lab/khmer/pull/1186/files#r35438920>.
>